### PR TITLE
Allow multiple locations in a single isochrone query

### PR DESCRIPTION
**Pull request by acwilton**
# Changes
+ Config changes have been made to allow up to 25 locations in a single request.
+ A new max_distance parameter has been added to isochrone under service_limits in valhalla_build_config. The default is set to 25 km. This may be increased in the future.
+ Minor tweak to the ConstructIsoTile function in isochrone where the grid is shifted to center around the center most location in the request instead of the first location in the request.

### CODE REVIEW

File: scripts/valhalla_build_config [hunk 1]
```diff
@@ -403,6 +404,7 @@ help_text = {
     'isochrone': {
       'max_contours': 'Maximum number of input contours to allow',
       'max_time': 'Maximum time value for any one contour',
+      'max_distance':'Maximum b-line distance between all locations in meters',
```
Comment (dnesbitt61) [inline scripts/valhalla_build_config]: "beeline" ?  I would just say "maximum distance between any two locations"

File: src/loki/isochrone_action.cc [hunk 2]
```diff
@@ -46,6 +64,12 @@ namespace valhalla {
       if (locations.size() > max_locations.find("isochrone")->second)
         throw valhalla_exception_t{400, 150, std::to_string(max_locations.find("isochrone")->second)};
 
+      //check the distances
+      auto max_location_distance = std::numeric_limits<float>::min();
+      check_distance(locations, max_distance.find("isochrone")->second, max_location_distance);
+      if (!healthcheck)
+        valhalla::midgard::logging::Log("max_location_distance::" + std::to_string(max_location_distance * kKmPerMeter) + "km", " [ANALYTICS] ");
```
Comment (kevinkreiser) [inline src/loki/isochrone_action.cc]: do we need to actually pick this up in logstash @kdiluca ? we dont use that much so maybe its not worth the effort at this point?

File: src/loki/service.cc [hunk 3]
```diff
@@ -241,7 +241,7 @@ namespace valhalla {
           continue;
         if (kv.first != "skadi" && kv.first != "trace")
           max_locations.emplace(kv.first, config.get<size_t>("service_limits." + kv.first + ".max_locations"));
-        if (kv.first != "skadi" && kv.first != "isochrone")
```
Comment (kevinkreiser) [inline src/loki/service.cc]: removing special cases feels good

File: src/thor/isochrone.cc [hunk 4]
```diff
@@ -83,8 +83,29 @@ void Isochrone::ConstructIsoTile(const bool multimodal, const unsigned int max_m
...
+  // Form bounding box that's just big enough to surround all of the locations.
+  PointLL center_ll = origin_locations[0].latlng_;
+  AABB2<PointLL> loc_bounds(center_ll.lng(), center_ll.lat(), center_ll.lng(), center_ll.lat());
+
+  for (const auto& loc : origin_locations) {
+    loc_bounds.Expand(loc.latlng_);
+  }
+
+  // Find the location closest to the center.
+  PointLL bounds_center = loc_bounds.Center();
```
Comment (kevinkreiser) [inline src/thor/isochrone.cc]: this is not the geometric center of the box? specifically which means the lat of this coord will be shifted away from the center as this box moves towards the poles. im not sure what effect that will have on isochrones but we can fix it by fixing `AABB2::Center` to use the midpoint algorithm of two diagonal corners. maybe it doesnt matter though

Comment (kevinkreiser) [inline src/thor/isochrone.cc]: actually maybe im wrong about this... lines of lat are spaced evenly and even though longitudes arent, the center of two lines of longitude will be the same despite changes in latitude. so yeah, i think im wrong about this?

File: scripts/valhalla_build_config (hunk 1 above)
Comment (acwilton) [inline scripts/valhalla_build_config]: That's how all of the other descriptions of max_distance are described so I just continued the pattern.

**Pull request merged**